src/nupic/types/Types.h

```diff
@@ -36,9 +36,10 @@
 
 #if defined(NTA_OS_WINDOWS) && defined(NTA_COMPILER_MSVC) && defined(NDEBUG)
 #pragma warning( disable : 4244 ) // conversion from 'double' to 'nta::Real', possible loss of data (LOTS of various type combinations)
-#pragma warning( disable : 4305 ) // truncation from 'double' to 'nta::Real', possible loss of data (LOTS of various type combinations)
 #pragma warning( disable : 4251 ) // needs to have dll-interface to be used by clients of class 
 #pragma warning( disable : 4275 ) // non dll-interface struct used as base for dll-interface class
+#pragma warning( disable : 4305 ) // truncation from 'double' to 'nta::Real', possible loss of data (LOTS of various type combinations)
+#pragma warning( once : 4838 ) // narrowing conversions
```

Comment:
4305 and 4244 covers very similar things, similar to 4838. These pagmas have mainly been used to reduce the build output. Any numbers destined for a float container, but don't have the 'f' suffix is treated as a double, and then can try and suggest multiple STL riddled pages of what to try.. I put 4838 as a once report (per file), to make sure that not all the warnings are suppressed.